core/src/tile/tileWorker.cpp

```diff
@@ -112,23 +112,63 @@ void TileWorker::run(Worker* instance) {
             continue;
         }
 
-        auto tileData = task->source().parse(*task, *builder->scene().mapProjection());
+        if (!task->hasTile()) {
 
-        // const clock_t begin = clock();
+            auto tileData = task->source().parse(*task, *builder->scene().mapProjection());
 
-        if (tileData) {
+            // const clock_t begin = clock();
 
-            auto tile = builder->build(task->tileId(), *tileData, task->source());
+            if (tileData) {
+
+                auto tile = builder->build(task->tileId(), *tileData, task->source());
+                // move tile to task (probably done if no rasters)
+                task->setTile(std::move(tile));
+
+            } else {
+                task->cancel();
+                m_pendingTiles = true;
+                requestRender();
+                continue;
+            }
 
             // float loadTime = (float(clock() - begin) / CLOCKS_PER_SEC) * 1000;
             // LOG("loadTime %s - %f", task->tile()->getID().toString().c_str(), loadTime);
+        }
 
-            // Mark task as ready
-            task->setTile(std::move(tile));
+        bool rastersReady = true;
+
+        if (task->source().rasterSources().size() != task->rasterTasks().size()) {
+            rastersReady = false;
         } else {
-            task->cancel();
+            for (auto& rasterTask : task->rasterTasks()) {
+                if (!rasterTask->hasRaster()) {
+                    rastersReady = false;
+                    break;
+                }
+            }
+        }
+
+        if (!rastersReady) {
+            // enqueue this task again
+            enqueue(std::move(task));
+            continue;
```

Comment:
One option seems to be to renqueue the dependent-tileTask once all the rasters are loaded: Instead of having a fixed ```m_rasterCallback``` one could pass a lambda as callback that binds the dependent task